### PR TITLE
AVM2: Implement escape() toplevel

### PR DESCRIPTION
**Pull request by CUB3D**
This implements the ```escape()``` toplevel function in as3, as per the [docs](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/package.html#escape())

### CODE REVIEW

File: core/src/avm2/globals/toplevel.rs [hunk 1]
```diff
@@ -349,3 +350,30 @@ pub fn parse_float<'gc>(
...
+        .unwrap_or(&Value::Undefined)
+        .coerce_to_string(activation)?;
+    let mut output = WString::new();
+
+    // Characters that are not escaped, sourced from as3 docs
+    let not_converted =
+        WStr::from_units(b"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ@-_.*+/");
+
+    for x in arg.iter() {
+        if not_converted.contains(WStr::from_units(&[x])) {
```
Comment (relrelb) [inline core/src/avm2/globals/toplevel.rs:370]: Doesn't simply `not_converted.contains(x)` work?

File: core/src/avm2/globals/toplevel.rs [hunk 2]
```diff
@@ -349,3 +350,30 @@ pub fn parse_float<'gc>(
...
+            output.push_str(WStr::from_units(&[x]));
```
Comment (relrelb) [inline core/src/avm2/globals/toplevel.rs:371]: Same here, doesn't simply `output.push(x)` work?

File: core/src/avm2/globals/toplevel.rs [hunk 3]
```diff
@@ -349,3 +350,34 @@ pub fn parse_float<'gc>(
 
     Ok(f64::NAN.into())
 }
+
+pub fn escape<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let output: AvmString = match args.first() {
```
Comment (relrelb) [inline core/src/avm2/globals/toplevel.rs:359]: nit: I think the type annotation here is unnecessary and can be omitted.

File: core/src/avm2/globals/toplevel.rs [hunk 4]
```diff
@@ -349,3 +350,34 @@ pub fn parse_float<'gc>(
...
+        None => "undefined".into(),
+        Some(Value::String(arg)) => {
```
[the remark this answers is not in the document]
Comment (CUB3D) [reply core/src/avm2/globals/toplevel.rs]: Done

**Review by relrelb — APPROVED** [review]

**Comment by relrelb** [comment]
Thank you!

**Pull request merged**